xtreme.rb

```diff
@@ -0,0 +1,123 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'zlib'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Xtreme Rat Controller Remote File Download Exploit',
+      'Description'    => %q{
+          This module exploits an arbitrary file download vulnerability in the Xtreme C&C server
+      },
+      'Author'         => 'Professor Plum',
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+        ],
+      'Platform'       => 'win',
+      'DisclosureDate' => 'Jul 27 2017',
+      'Targets'        =>
+        [
+          ['Xtreme RAT 3.6', { 'Ver' => '3.6' }],
+          ['Xtreme RAT 3.7', { 'Ver' => '3.7' }]
+        ],
+      'Privileged'     => false,
+      'DefaultTarget' => 1))
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETFILE', [false, 'Target file to download', 'user.info'])
```

Comment:
`TARGETFILE` should probably be `true` rather than `false` as a target file is required.